src/native-stack/types.tsx

```diff
@@ -222,6 +223,10 @@ export type NativeStackNavigationOptions = {
    * Boolean indicating whether the navigation bar is translucent.
    */
   headerTranslucent?: boolean;
+  /**
+   * A callback called every frame during the transition of screens of `native-stack`.
+   */
+  onTransitionProgress?: (e: TransitionProgressEvent) => void;
```

Comment:
IMO there shouldn't be such an option. We don't expose events in options, we have a separate event system for it. But it also probably shouldn't be a regular event coz then we would need to do animations in JS only. Instead we should expose an `Animated.Value` like JS stack does. JS stack has a `useCardAnimation()` hook that we can use in screens in stack. It has many properties in JS stack, but for native stack, this should be enough:

```js
{
  /**
   * Values for the current screen.
   */
  current: {
    /**
     * Animated node representing the progress value of the current screen.
     */
    progress: Animated.AnimatedInterpolation;
  };
  /**
   * Values for the screen after this one in the stack.
   * This can be `undefined` in case the screen animating is the last one.
   */
  next?: {
    /**
     * Animated node representing the progress value of the next screen.
     */
    progress: Animated.AnimatedInterpolation;
  };
};
```

We'd basically need to create animated values for each screen that we can update with `Animated.event` and `useNativeDriver: true`.